### ISSUE TITLE
ts types seem to be not correct - 9.0.9

### ISSUE THREAD

**Issue by RobbyRabbitman** [storybookjs/eslint-plugin-storybook#209]
**Describe the bug**

ts types seem to be not correct

**To Reproduce**

I have a shared eslint config in a monorepo written in ts und need to cast the default export of 'eslint-plugin-storybook'

```ts
import { type Linter } from 'eslint';
import storybookEslintPlugin from 'eslint-plugin-storybook';
import ignoreEslintConfig from './eslint-config-ignore.js';
import webEslintConfig from './eslint-config-web.js';

export const storybookEslintConfig = [
  ...ignoreEslintConfig,
  ...webEslintConfig,
  /** https://storybook.js.org/docs/configure/integration/eslint-plugin */
  ...(
    storybookEslintPlugin as unknown as typeof import('eslint-plugin-storybook').default
  ).configs['flat/csf-strict'],
  {
    ignores: ['!.storybook'],
  },
] satisfies Linter.Config[];

export default storybookEslintConfig;
```

The index.d.ts of the plugin has this weird export - does not seem to me like a export default but im not a ts pro.

```ts
export { _default as default };
```

**Comment by psychobolt** [comment]
Ran into the same issue. `as default` is not a actual syntax for default export but a variable declaration. It should be `export default _default;`

**Comment by aaronadamsCA** [comment]
https://arethetypeswrong.github.io/?p=eslint-plugin-storybook@9.1.13

<img width="1604" height="379" alt="The resolved types use export default where the JavaScript file appears to use module.exports =. This will cause TypeScript under the node16 module mode to think an extra .default property access is required, but that will likely fail at runtime. These types should use export = instead of export default." src="https://github.com/user-attachments/assets/bbccabdd-af08-46e7-af85-30646715e4ed" />
